crates/domains/src/infra/repository/query_builder.rs

```diff
@@ -136,36 +124,109 @@ pub trait QueryParamsBuilder {
         }
     }
 
+    /// Forms the required SQL clauses to replace the cursor with the values
+    ///
+    /// Example:
+    ///
+    ///  ```md
+    ///  block_height    transaction_index     receipt_index
+    ///
+    ///  0001                    0                 0
+    ///  0001                    0                 1     <---
+    ///  0001                    0                 2
+    ///  0001                    1                 0
+    ///  0001                    2                 0
+    ///  0002                    0                 0
+    ///  0002                    0                 1
+    ///  ```
+    ///
+    ///  ```sql
+    ///  WHERE (
+    ///    (block_height = 0001 AND transaction_index = 0 AND receipt_index > 1) OR
+    ///    (block_height = 0001 AND transaction_index > 0) OR
+    ///    (block_height > 0001)
+    ///  )
+    ///  ```
+    fn create_pagination_conditions(
+        cursor_fields: &[&str],
+        cursor: &Cursor,
+        operation: &str,
+    ) -> String {
+        if cursor_fields.is_empty() || cursor.is_empty() {
+            return String::new();
+        }
+
+        let cursor_values = cursor.split();
+
+        let result = (0..cursor_values.len())
+            .rev()
+            .map(|i| {
+                let equality_conditions = (0..i)
+                    .map(|j| {
+                        format!("{} = {}", cursor_fields[j], cursor_values[j])
+                    })
+                    .collect::<Vec<_>>();
+
+                let operation_condition = format!(
+                    "{} {} {}",
+                    cursor_fields[i], operation, cursor_values[i]
+                );
+
+                let mut all_conditions = equality_conditions;
+                all_conditions.push(operation_condition);
+
+                format!("({})", all_conditions.join(" AND "))
+            })
+            .collect::<Vec<_>>()
+            .join(" OR ");
+        format!("({})", result)
+    }
+
     fn apply_pagination(
         query_builder: &mut QueryBuilder<Postgres>,
         pagination: &QueryPagination,
-        cursor_field: &str,
-        join_prefix: Option<&str>,
```

Comment:
Can we remove the prefix here? I believe some places use it to create pagination over join query. Like predicates endpoint